app/activelayer.h

```diff
@@ -0,0 +1,64 @@
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+
+#ifndef ACTIVELAYER_H
+#define ACTIVELAYER_H
+
+#include <QObject>
+#include "qgsmaplayer.h"
+#include "qgsvectorlayer.h"
+
+
+/**
+ * @brief The ActiveLayer class holds information about current active layer for recording.
+ * It supports setting active layer by layer and offers information about layer.
```

Comment:
It would be useful to add a note that it is registered as `__activeLayer` object in QML ?